algoliasearch-common/src/main/java/com/algolia/search/AsyncObjects.java

```diff
@@ -206,7 +207,7 @@
    * @param objectID the unique identifier of the object to retrieve
    * @return the associated AsyncTask
    */
-  default CompletableFuture<AsyncTask> deleteObject(@Nonnull String objectID) {
+  default CompletableFuture<AsyncTask> deleteObject(@Nonnull String objectID) throws AlgoliaException {
```

Comment:
careful, you are changing the signature of the method, it's not backward compatible